amundsen_application/static/js/components/common/Bookmark/MyBookmarks/index.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { GlobalState } from "ducks/rootReducer";
+
+import './styles.scss'
+import { Bookmark } from "ducks/bookmark/types";
+import {
+  BOOKMARK_TITLE,
+  BOOKMARKS_PER_PAGE,
+  EMPTY_BOOKMARK_MESSAGE,
+  MY_BOOKMARKS_SOURCE_NAME,
+} from "./constants";
+import ResourceList from "components/common/ResourceList";
+
+interface StateFromProps {
+  myBookmarks: Bookmark[];
+  isLoaded: boolean;
+}
+
+export type MyBookmarksProps = StateFromProps;
+
+interface MyBookmarksState {
+  activePage: number;
+}
+
+export class MyBookmarks extends React.Component<MyBookmarksProps, MyBookmarksState> {
```

Comment:
Why the rename? Is there some other component that you imagine being called `BookmarkList`? I think `BookmarkList` is a more appropriate name because it names the component simply based on what it is, without any further implications. 

Is the rename because we anticipate having logic that will apply based on whether or not this bookmark list is for the current user vs another user? If so, then I'd propose that the bookmark list component have whatever boolean properties it needs to do any conditional rendering. Let me know your thoughts.

Reply:
In my head the `ProfilePage` will be responsible for fetching bookmarks, owned, and frequently used resources and put them into a `ResourceList` without making separate components for each of them.

```
class ProfilePage {
  ... 
  componentDidMount() {
    this.props.getBookmarksForUser(userId);
    this.props.getFrequentlyUsedForUser(userId);
    this.props.getOwnedForUser(userId);
  }

  getTabsInfo() {
    return [ 
       { content: <ResouceList items={ this.props.owned } ...>, title: "owned", }
       { content: <ResouceList items={ this.props.bookmarks } ...>, title: "bookmarks" }
       { content: <ResouceList items={ this.props.frequentlyUses } ...>, title: "frequently uses" }
    ]
  }
}
```

In this case we only need one `MyBookmarks` component which connects the `globalState.bookmarks.myBookmarks`. I'm also removing the state management from this component so it should be a relatively light weight.

Reply:
In an alternative version of this, we would create 3 very similar components that hit different endpoints and are responsible for fetching their own data. At this point I don't think it's worth creating a separate component for each tab, but I may change my mind.

Reply:
I see. This resolves my concern of whether or not there would be some duplicate bookmarks component. Still not attached to the name `MyBookmarks`, but we have some other questionable names and at least this one is tied to the piece of globalState that it uses. Thanks for the explanation.